app/controllers/schedules_controller.rb

```diff
@@ -1,20 +1,54 @@
 class SchedulesController < ApplicationController
+  before_action :check_string, only: [:create]
+
   def show
     @shifts = []
     location = Location.find(params[:location_id])
     location.users.each do |user|
       @shifts << Array(user_shifts(user))
     end
+    @date = String.new
+  end
+
+  def create
+    @shifts = []
+    location = Location.find(params[:location_id])
+    location.users.each do |user|
+      @shifts << Array(user_shifts(user))
+    end
+    @date = String.new
+
+    render :show
   end
 
   private
 
+  def week_date
+    params[:dates][:date].gsub(/\D/,'/')
+  end
+
+  def date_to_use
+    if params[:dates] == nil
+      DateTime.now.all_week
+    else
+      range_of_dates
+    end
+  end
+
+  def range_of_dates
+    convert_date(params[:dates][:date].gsub(/\D/,'/')).all_week
```

Comment:
Space missing after comma.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.